src/lib.rs

```diff
@@ -357,7 +261,7 @@ impl SecretKey {
     ///
     /// *WARNING* this constructor will overwrite the referenced `Fr` element with zeros after it
     /// has been copied onto the heap.
-    pub fn from_mut(fr: &mut Fr) -> Self {
+    pub fn from_mut(fr: &mut blst_fr) -> Self {
```

Comment:
Do we need to expose blst_fr in public API?    Can we not just `pub type Fr = blst_fr` or use a wrapper struct to keep the Fr impl abstract?

Also, about naming, shouldn't the type be BlstFr rather than blst_fr?

Reply:
Both sn_dbc (both forks) and blsbs are using Fr/IntoFr.   So just from a "least breakage" standpoint I'd prefer to keep the old naming unless there's a good reason not to.

> So it looks like it won't be many changes.

yeah, so it seems a smaller/easier change there than in the dependent crates.

> Do we want to pretend to be bls12_381::Fr or should we keep it as blst::blst_fr?

I'm ok with pretending/abstracting.  I guess I see it as blsttc uses/provides an Fr, and dependent crate shouldn't really need to know or care that it is Blst or bls12_381.

> It's called blst_fr and not BlstFr because it comes from this

I'm no rust style expert myself.  But It appears to me that blst is keeping the c/c++ naming convention because it is a rust binding around a c/c++ library.  Stylistically it is odd because afaik rust convention for data type names is camel-case.  I'm curious, does cargo fmt and/or clippy not complain?

Reply:
> I wonder if we should do the same for blst_p1 as G1 and blst_p2 as G2?

I would think so, for consistency.

G2 is presently used a bit in blsbs, but not in sn_dbc.   And maybe/likely the blsbs usage can go away with your latest changes anyway, I'm not sure.